### PR TITLE
CI: add git commit message linter

### LINKED ISSUE #1231: CI should check that commits are signed
Ensure that commits that make up a PR are signed (`-s`) .
Use as [inspiration](https://github.com/firecracker-microvm/firecracker-containerd/blob/2905f8dbab16c6847c99dd6384a90b462bffe206/.buildkite/logcheck.sh#L31).

### PR DESCRIPTION
**Pull request by dianpopa**
## Reason for This PR

Fixes #1231

## Description of Changes

* install [gitlint](https://jorisroovers.com/gitlint/rules/) in the x86_64 docker image
* create `.gitlint` config file containing 2 rules: title-max-length and body-max-length
* create 1 additional [user defined rule](https://jorisroovers.com/gitlint/user_defined_rules/) for checking against "Signed-off-by" message 

- [ ] This functionality can be added in [`rust-vmm`](https://github.com/rust-vmm).

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The reason for this PR is clearly provided (issue no. or explanation).
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] Any newly added `unsafe` code is properly documented.
- [x] Any API changes are reflected in `firecracker/swagger.yaml`.
- [x] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.

### CODE REVIEW

File: tests/framework/gitlint_rules.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+from gitlint.rules import CommitRule, RuleViolation
+
+
+class SignedOffBy(CommitRule):
+    """Make sure that each commit contains a "Signed-off-by" line."""
+
+    # The name of the rule.
+    name = "body-requires-signed-off-by"
+
+    # The unique if of the rule.
```
Comment (luminitavoicu) [inline tests/framework/gitlint_rules.py:14]:
```suggestion
    # The unique id of the rule.
```

**Review by luminitavoicu — COMMENTED** [review]
Commit looks good, just one little comment.

File: tests/framework/gitlint_rules.py (hunk 1 above)
Comment (dianpopa) [reply tests/framework/gitlint_rules.py]: Done!

**Review by luminitavoicu — APPROVED** [review]

**Review by AlexandruCihodaru — APPROVED** [review]

**Comment by dianpopa** [comment]
Closing in favor of #2388.

**Pull request closed without merging**